stock_card_report/reports/stock_card_report.py

```diff
@@ -0,0 +1,109 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockCardView(models.TransientModel):
+    _name = 'stock.card.view'
+    _description = 'Stock Card View'
+    _order = 'date'
+
+    date = fields.Datetime()
+    product_id = fields.Many2one(comodel_name='product.product')
+    product_qty = fields.Float()
+    product_uom_qty = fields.Float()
+    product_uom = fields.Many2one(comodel_name='uom.uom')
+    reference = fields.Char()
+    location_id = fields.Many2one(comodel_name='stock.location')
+    location_dest_id = fields.Many2one(comodel_name='stock.location')
+    is_initial = fields.Boolean()
+    product_in = fields.Float()
+    product_out = fields.Float()
+
+
+class StockCardReport(models.TransientModel):
+    _name = 'report.stock.card.report'
+    _description = 'Stock Card Report'
+
+    # Filters fields, used for data computation
+    date_from = fields.Date()
+    date_to = fields.Date()
+    product_ids = fields.Many2many(
+        comodel_name='product.product',
+    )
+    location_id = fields.Many2one(
+        comodel_name='stock.location',
+    )
+
+    # Data fields, used to browse report data
+    results = fields.Many2many(
+        comodel_name='stock.card.view',
+        compute='_compute_results',
+        help='Use compute fields, so there is nothing store in database',
+    )
+
+    @api.multi
+    def _compute_results(self):
+        self.ensure_one()
+        change = False
+        if not self.date_from:
+            change = True
+            self.date_from = '0001-01-01'
+        if not self.date_to:
+            self.date_to = fields.Date.context_today(self)
+        locations = self.env['stock.location'].search(
+            [('id', 'child_of', [self.location_id.id])])
+        self._cr.execute("""
+            SELECT move.date, move.product_id, move.product_qty,
+                move.product_uom_qty, move.product_uom, move.reference,
+                move.location_id, move.location_dest_id,
+                case when move.location_dest_id in %s
+                    then move.product_qty end as product_in,
+                case when move.location_id in %s
+                    then move.product_qty end as product_out,
+                case when move.date < %s then True else False end as is_initial
+            FROM stock_move move
+            WHERE (move.location_id in %s or move.location_dest_id in %s)
+                and move.state = 'done' and move.product_id in %s
+                and CAST(move.date AS date) <= %s
+            ORDER BY move.date, move.reference
+        """, (
+            tuple(locations.ids), tuple(locations.ids), self.date_from,
+            tuple(locations.ids), tuple(locations.ids),
+            tuple(self.product_ids.ids+[0]), self.date_to))
+        stock_card_results = self._cr.dictfetchall()
+        ReportLine = self.env['stock.card.view']
+        for line in stock_card_results:
```

Comment:
```suggestion
        self.results = [ReportLine.new(line).id for line in stock_card_results]
```
performance faster than a little bit

Reply:
Changed.